src/server/main_service.cc

```diff
@@ -2190,14 +2164,23 @@ void Service::Exec(CmdArgList args, const CommandContext& cmd_cntx) {
         BorrowedInterpreter(cmd_cntx.tx, &cntx->conn_state).Release();
   }
 
+  auto keys = CollectAllKeys(&exec_info);
+  UniqueSlotChecker slot_checker;
```

Comment:
lets do this only in cluster mode
and move it up before DetermineScriptPresense